enhancements/kueue/openshift-kueue.md

```diff
@@ -0,0 +1,749 @@
+	// resources provides additional configuration options for handling the resources.
+	// Supports https://github.com/kubernetes-sigs/kueue/blob/release-0.10/keps/2937-resource-transformer/README.md
+	// +optional
+	Resources *kueuecfgv1beta1.Resources `json:"resources,omitempty"`
+	// ManageJobsWithoutQueueName controls whether or not Kueue reconciles
+	// jobs that don't set the annotation kueue.x-k8s.io/queue-name.
+	// Allowed values are NoQueueName and QueueName
+	// Default will be QueueName
+	// +optional
+	ManageJobsWithoutQueueName *ManageJobsWithoutQueueNameOption `json:"manageJobsWithoutQueueName,omitempty"`
+	// ManagedJobsNamespaceSelector can be used to omit some namespaces from ManagedJobsWithoutQueueName
+	// +optional
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
+	// FairSharing controls the fair sharing semantics across the cluster.
+	FairSharing *kueuecfgv1beta1.FairSharing `json:"fairSharing,omitempty"`
+}
+
+// KueueStatus defines the observed state of Kueue
+type KueueStatus struct {
+	operatorv1.OperatorStatus `json:",inline"`
+}
+```
+
+We will use the kueue configuration [API](https://github.com/kubernetes-sigs/kueue/blob/main/apis/config/v1beta1/configuration_types.go).
+
+The operator reads the kueue configuration and generates a ConfigMap that the kueue manager deployment will use. 
+We will only expose a limited amount of configurations.
+
+The flow is as follows:
+
+1) User specifies a Kueue CRD
+1) Configuration is specified
+1) A Configmap will be created with the configuration specified
+1) Kueue deployment will use that configmap in the operand deployment.
+
+The operator will manage these components necessary for Kueue. 
+- Custom Resources
+- Cluster Roles
+- Cluster Role Bindings
+- Services
+- Mutating Webhook
+- Validating Webhook
+- ServiceMonitors
+- APIService for visibility
+- Roles
+- Certificates
+- Issuers
+
+The operator will create and apply these resources to the cluster.
+
+The operator has the following requirements.
+
+a. Konflux integration
+b. Support for x64 and ARM
+c. Disconnected
+d. FIPS
+
+The operator will be OLM managed and hosted on OperatorHub.
+
+### Workflow Description
+
+#### Batch Job Administrator
+
+As a admin, I want to enable Kueue to manage batch jobs.
+
+```yaml
+apiVersion: operator.openshift.io/v1beta1
+kind: Kueue
+metadata:
+ name: cluster
+ namespace: openshift-kueue-operator
+spec:
+   config:
+     integrations:
+       frameworks:
+         - "batch/job"
+```
+
+This will create a Kueue deployment that will manage batch jobs. 
+
+#### RHOAI Enablement
+
+Red Hat Openshift AI is already using Kueue in production. Their deployment can be replicated:
+
+```yaml
+apiVersion: operator.openshift.io/v1beta1
+kind: Kueue
+metadata:
+ labels:
+ name: cluster
+ namespace: openshift-kueue-operator
+spec:
+  config:
+     integrations:
+      frameworks:
+       - "batch/job"
```

Comment:
Just curious for my understanding - does OLM v1 currently also create jobs to unpack bundles and install. We frequently were having issues with OLM tending to be stuck during upgrades, as Kueue webhooks stopped them from getting created. There is a fix for this that was added. Just want to know if it would still remain an issue with v1.

Reply:
That is good to know! Thanks joe.